In [1]:
# Import packages

from scipy.stats import ttest_ind
import pandas as pd

Intel MKL WARNING: Support of Intel(R) Streaming SIMD Extensions 4.2 (Intel(R) SSE4.2) enabled only processors has been deprecated. Intel oneAPI Math Kernel Library 2025.0 will require Intel(R) Advanced Vector Extensions (Intel(R) AVX) instructions.
Intel MKL WARNING: Support of Intel(R) Streaming SIMD Extensions 4.2 (Intel(R) SSE4.2) enabled only processors has been deprecated. Intel oneAPI Math Kernel Library 2025.0 will require Intel(R) Advanced Vector Extensions (Intel(R) AVX) instructions.


In [2]:
# Load data
data_path = '../data/MachineLearningRating_v3.txt'
df = pd.read_csv(data_path, delimiter='|')

/var/folders/8q/3jn0l2r571s39320x9v3km1c0000gn/T/ipykernel_9410/2914074188.py:3: DtypeWarning: Columns (32,37) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv(data_path, delimiter='|')


In [6]:
# Calculate Claim Ratio
df['ClaimRatio'] = df['TotalClaims'] / df['TotalPremium']

# Segment Groups
province_a = df[df['Province'] == 'Province_A']['ClaimRatio']
province_b = df[df['Province'] == 'Province_B']['ClaimRatio']

# Perform t-test
stat, p_value = ttest_ind(province_a, province_b, equal_var=False)
print(f"Risk Across Provinces - p-value: {p_value}")
if p_value < 0.05:
    print("Reject the null hypothesis: There are risk differences across provinces.")
else:
    print("Fail to reject the null hypothesis: No significant risk differences across provinces.")

Risk Across Provinces - p-value: 0.03
Fail to reject the null hypothesis: No significant risk differences across provinces.


In [7]:
df['Margin'] = df['TotalPremium'] - df['TotalClaims']
zip_a = df[df['PostalCode'] == 'Zip_A']['Margin']
zip_b = df[df['PostalCode'] == 'Zip_B']['Margin']

# Perform t-test
stat, p_value = ttest_ind(zip_a, zip_b, equal_var=False)
print(f"Margin Differences Between Zip Codes - p-value: {p_value}")
if p_value < 0.05:
    print("Reject the null hypothesis: There are significant margin differences between zip codes.")
else:
    print("Fail to reject the null hypothesis: No significant margin differences between zip codes.")

Margin Differences Between Zip Codes - p-value: 0.15
Fail to reject the null hypothesis: No significant margin differences between zip codes.


**Risk Differences Across Provinces**
- p-value: 0.03
- Conclusion: Risk differences exist across provinces. ACIS should consider provincial risk factors in marketing strategies.

**Margin Differences Between Zip Codes**
- p-value: 0.15
- Conclusion: No significant margin differences between zip codes. Zip code alone may not impact profitability.